edflow/explore.py

```diff
@@ -48,23 +62,154 @@ def display(key, obj):
+    """Displays custom visualizations in streamlit
+
+    The visualizations can be inserted to the config via their import path.
+    Everyone can implement a custom visualization for an example.
+
+    The visualization functions must accept the example and config as positional
+    arguments.
+
+
+    Examples
+    --------
+
+    Add visualizations to the text box with their import path. For example:
+
+    .. code-block:: python
+
+        edflow.util.edexplore.display_flow_on_image
+
+
+    A valid visualization function could look like for example:
+
+    .. code-block:: python
+
+        import streamlit as st
+        from edflow.util.edexplore import isimage, st_get_list_or_dict_item
+
+        def my_visualization(ex, config):
+            st.write("This is my visualization")
+
+            image1, image1_key = st_get_list_or_dict_item(ex, "image1", filter_fn=isimage)
+
+            st.image((image1 + 1) / 2)
+            
+            image2 = ex["image2"]
+            image3 = ex["image3"]
+
+            st.image((image2 + 1) / 2)
+            st.image((image3 + 1) / 2)
+
+
+    Visualizations can be displayed by default, if they are specified in the config.
+    An example for the configuration yaml file would be:
+
+    .. code-block::
+
+        edexplore:
+            custom_visualizations:
+                my_visualization_name:
+                    path: my_package.visualizations.my_visualization
+
+
+    Parameters
+    ----------
+    ex : dict
+        Example to be visualized
+    config : dict
+        Edexplore config
+    """
+    st.header("Custom visualizations")
+    default_visualizations = retrieve(
+        config, "edexplore/custom_visualizations", default=dict()
+    )
+    import_paths = [
+        vis.get("path", None)
+        for vis in default_visualizations.values()
+        if isinstance(vis.get("path", None), str)
+    ]
+    visualizatins_str = st.text_input(
+        "Visualization import paths: comma separated", value=",".join(import_paths)
+    )
+
+    if len(visualizatins_str) == 0:
+        st.text(custom_visualizations.__doc__)
+
+    for vis in [vis for vis in visualizatins_str.split(",") if vis != ""]:
+        try:
+            st.subheader(vis)
+            impl = get_obj_from_str(vis, reload=True)
+            impl(ex, config)
+        except Exception as error:
+            st.write(error)
+
+
+ADDITIONAL_VISUALIZATIONS = {
+    "custom_visualizations": custom_visualizations,
+    "optical_flow_on_image": display_flow_on_image,
+}
+
+
+def show_example(dset: DatasetMixin, idx: int, config: dict) -> None:
+    """Show example of dataset
+
+    Parameters
+    ----------
+    dset : DatasetMixin
+        Dataset to be shown
+    idx : int
+        Index to be shown
+    config : dict
+        Config used to show example
+    """
+
     ex = dset[idx]
+
+    # additional visualizations
+    default_additional_visualizations = retrieve(
+        config, "edexplore", default=dict()
```

Comment:
How should the edexplore config be at best?
Right now it is like:

```
edexplore:
  optical_flow_on_image:
    image_key: "image1"
  custom_visualizations:
    vis1:
      path: "my.import.path"
    vis2:
      path: "my.import.path2"
```

Maybe something like:
```
edexplore:
  visualizations:
    optical_flow_on_image:
      image_key: "image1"
    custom:
      vis1:
        path: "my.import.path"
      vis2:
        path: "my.import.path2"
```
would be better, because then edexplore can have also other config parameters for future expansion of the module.

Reply:
both are fine but I would probably also prefer the latter.

Reply:
changed it to the latter